sdk/storage/azure-storage-file-datalake/src/main/java/com/azure/storage/file/datalake/models/PathProperties.java

```diff
@@ -107,116 +110,123 @@ public PathProperties(final OffsetDateTime creationTime, final OffsetDateTime la
         this.encryptionKeySha256 = encryptionKeySha256;
         this.accessTierChangeTime = accessTierChangeTime;
         this.metadata = metadata;
+        /* Default isDirectory to false. */
+        if (this.metadata == null) {
+            this.isDirectory = false;
+        } else {
+            this.isDirectory = String.valueOf(true)
+                .equalsIgnoreCase(metadata.get(Constants.HeaderConstants.DIRECTORY_METADATA_KEY));
```

Comment:
Mind replacing this with `Boolean.parseBoolean(metadata.get(Constants.HeaderConstants.DIRECTORY_METADATA_KEY))`, it does the same thing.

Reply:
done